pkg/investigations/diagnosticcollection/diagnosticcollection.go

```diff
@@ -0,0 +1,141 @@
+	"github.com/openshift/configuration-anomaly-detection/pkg/investigations/diagnosticcollection/inspect"
+	"github.com/openshift/configuration-anomaly-detection/pkg/investigations/investigation"
+	"github.com/openshift/configuration-anomaly-detection/pkg/logging"
+	"github.com/openshift/configuration-anomaly-detection/pkg/notewriter"
+)
+
+type Investigation struct{}
+
+func (c *Investigation) Run(rb investigation.ResourceBuilder) (investigation.InvestigationResult, error) {
+	result := investigation.InvestigationResult{}
+
+	// Build resources with K8s client (needed for oc commands)
+	r, err := rb.WithK8sClient().Build()
+	if err != nil {
+		return result, fmt.Errorf("failed to build resources: %w", err)
+	}
+
+	// Initialize PagerDuty note writer
+	notes := notewriter.New(r.Name, logging.RawLogger)
+	defer func() { r.Notes = notes }()
+
+	// Get the alert name from the investigation name
+	// In a real scenario, this would come from the PagerDuty webhook payload
+	// For now, we'll use a default or extract from context
+	alertName := "UpgradeConfigSyncFailureOver4HrSRE" // TODO: Get from payload
+
+	// Find mapping for this alert
+	mapping := GetMappingForAlert(alertName)
+	if mapping == nil {
+		notes.AppendWarning("No diagnostic mapping found for alert: %s", alertName)
+		return result, r.PdClient.EscalateIncidentWithNote(notes.String())
+	}
+
+	logging.Infof("Using mapping: %s - %s", mapping.AlertPattern, mapping.Description)
+	notes.AppendAutomation("Collecting diagnostics: %s", mapping.Description)
+
+	// Create inspect executor
+	executor := inspect.New(r.K8sClient)
+	defer func() {
+		// Cleanup will be called even if we return early
+	}()
+
+	// Run oc adm inspect
+	inspectDir, err := executor.Execute(mapping.Resources)
+	if err != nil {
+		notes.AppendWarning("Failed to collect diagnostics: %v", err)
+		logging.Errorf("oc adm inspect failed: %v", err)
+		return result, r.PdClient.EscalateIncidentWithNote(notes.String())
+	}
+	defer executor.Cleanup(inspectDir)
+
+	notes.AppendSuccess("Diagnostic data collected successfully")
+	logging.Infof("Diagnostics collected in: %s", inspectDir)
+
+	// Run analyzers
+	allFindings := findings.New()
+
+	// Determine which analyzers to run based on collected resources
+	analyzerList := c.getAnalyzersForResources(mapping.Resources)
+
+	for _, analyzer := range analyzerList {
+		logging.Infof("Running analyzer: %s", analyzer.Name())
+
+		analyzerFindings, err := analyzer.Analyze(inspectDir)
+		if err != nil {
+			logging.Warnf("Analyzer %s failed: %v", analyzer.Name(), err)
+			notes.AppendWarning("Analysis failed for %s: %v", analyzer.Name(), err)
+			continue
+		}
+
+		// Merge findings
+		for _, finding := range analyzerFindings.GetAll() {
+			allFindings.Add(finding)
+		}
+	}
+
+	// Format findings for PagerDuty
+	if !allFindings.IsEmpty() {
+		notes.AppendAutomation("Diagnostic Analysis Results (%d findings)", allFindings.Count())
```

Comment:
Can we append files to notes?  It would be pretty awesome to include the entire inspect-dir content.

Reply:
We cannot today, but once I finish implementing https://issues.redhat.com/browse/SREP-2137, we will be able to store (for now) text data related to clusters, and link to it from PagerDuty.